dev-docs/RFCs/v4.0.0/names_for_concepts_types_functions.md

```diff
@@ -0,0 +1,225 @@
+| `hexRange`   | `gridDiskUnsafe`                         | `hexRangeDistances`, does not allocate distances |
+| *DNE*        | `gridDiskSafe`                           |                                                  |
+| `kRing`      | `gridDisk`                               | `kRingDistances`, allocates and drops distances  |
+| `hexRanges`  | `gridDiskMultiUnsafe`, `gridDisksUnsafe` | N x `hexRange`                                   |
+
+- **TODO**: Do we *really* want to keep `hexRanges` in the API? It sounds like it currently has applications, but how
+  hard would it really be for a user to reproduce this functionality by just calling `hexRange` multiple times, and moving
+  the pointer before each call? It is also weird that we only provide the "unsafe" version of this function in the API.
+- **Idea**: leave it out initially, and see if anyone complains? It would be very easy to add back in.
+
+#### Hollow Ring
+
+| Current name |  Proposed name   | Calls |
+|--------------|------------------|-------|
+| `hexRing`    | `gridRingUnsafe` | NONE  |
+| *DNE*        | `gridRingSafe`   |       |
+| *DNE*        | `gridRing`       |       |
+
+
+### H3 Edge Types
+
+Instead of `UnidirectionalEdge`, use the term `DirectedEdge`.
+
+For a future undirected edge mode, use the term `Edge`.
+
+|                  Current name                 |        Proposed name         |
+|-----------------------------------------------|------------------------------|
+| `h3UnidirectionalEdgeIsValid`                 | `isValidDirectedEdge`        |
+| `getH3UnidirectionalEdge`                     | `cellsToDirectedEdge`        |
+| `getH3IndexesFromUnidirectionalEdge`          | `directedEdgeToCells`        |
+| `getH3UnidirectionalEdgesFromHexagon`         | `originToDirectedEdges`      |
+| *DNE*                                         | `destinationToDirectedEdges` |
+| `getH3UnidirectionalEdgeBoundary`             | `directedEdgeToLine`         |
+| `getOriginH3IndexFromUnidirectionalEdge`      | `getDirectedEdgeOrigin`      |
+| `getDestinationH3IndexFromUnidirectionalEdge` | `getDirectedEdgeDestination` |
+
+todo: happy with `directedEdgeToLine`?
+
+
+### Area/Length Functions
+
+|  Current name  |        Proposed name        |         Notes         |
+|----------------|-----------------------------|-----------------------|
+| `hexAreaKm2`   | `getHexagonAreaAvgKm2`      |                       |
+| `hexAreaM2`    | `getHexagonAreaAvgM2`       |                       |
+| `edgeLengthKm` | `getHexagonEdgeLengthAvgKm` |                       |
+| `edgeLengthM`  | `getHexagonEdgeLengthAvgM`  |                       |
+| *DNE*          | `getPentagonAreaAvg*`       |                       |
+| *DNE*          | `getPentagonEdgeLengthAvg*` |                       |
+| *DNE*          | `cellAreaKm2`               | area of specific cell |
+| *DNE*          | `cellAreaM2`                | area of specific cell |
```

Comment:
Is everyone happy with these changes to the area/length functions?

Reply:
👍